File: examples/acoustics_1d_example1/fortran/setprob.f90
subroutine setprob

    implicit none
    character*25 :: fname
    integer :: iunit
    real(kind=8) :: rho,bulk,cc,zz,beta

    common /cparam/ rho,bulk,cc,zz   
    common /cqinit/ beta
 
    ! Set the material parameters for the acoustic equations
    ! Passed to the Riemann solver rp1.f in a common block
 
    iunit = 7
    fname = 'setprob.data'
    ! open the unit with new routine from Clawpack 4.4 to skip over
    ! comment lines starting with #:
    call opendatafile(iunit, fname)


    ! density:
    read(7,*) rho

    ! bulk modulus:
    read(7,*) bulk

    ! sound speed:
    cc = dsqrt(bulk/rho)

    ! impedance:
    zz = cc*rho

    ! beta for initial conditions:
    read(7,*) beta

end subroutine setprob

File: examples/acoustics_1d_example1/SetProb.jl

module SetProb

export
    CParam, read_prob_data

type CParam
    rho   ::Float64
    bulk  ::Float64
    cc    ::Float64
    zz    ::Float64

    beta  ::Float64

    # default constructor
    function CParam()
        new(0.0, 0.0, 0.0, 0.0, 0.0)
    end
end


# Set the material parameters for the acoustic equations

using ReadDataFile

function read_prob_data(parms::CParam)

    r = Reader("setprob.data")

    # density:
    parms.rho  = readflt(r)

    # bulk modulus:
    parms.bulk = readflt(r)

    # sound speed:
    parms.cc = sqrt(parms.bulk/parms.rho)

    # impedance:
    parms.zz = parms.cc*parms.rho

    # beta for initial conditions:
    parms.beta = readflt(r)

    close_reader(r)
end # read_prob_data

end # SetProb

using .SetProb
using Base.Test.@test

function test_read_data()
    const parms = CParam()
    read_prob_data(parms)
    println("CParam = $parms")
    return true
end

# @test test_read_data()


